examples/with-next-auth/pages/api/users/[[...params]].ts

```diff
@@ -0,0 +1,48 @@
+import {
+  Get,
+  Query,
+  createHandler,
+  ParseNumberPipe,
+  DefaultValuePipe,
+  createMiddlewareDecorator,
+  NextFunction,
+  UnauthorizedException,
+  InternalServerErrorException,
+} from '@storyofams/next-api-decorators';
+import { NextApiRequest, NextApiResponse } from 'next';
+import { getToken } from 'next-auth/jwt';
+import { User, sampleUserData } from '../../../data';
+
+declare module 'next' {
+  interface NextApiRequest {
+    user?: { name: string }
+  }
+}
+
+const NextAuthGuard = createMiddlewareDecorator(async (req: NextApiRequest, _res: NextApiResponse, next: NextFunction) => {
+  try {
+    const token = await getToken({ req, secret: process.env.JWT_SECRET });
+    if (!token || !token.name) {
+      throw new UnauthorizedException();
```

Comment:
This will always throw a `InternalServerErrorException` because it's in a try/catch.

Reply:
Pff, true. Thanks for the catch!